servers/zms/src/main/java/com/yahoo/athenz/zms/provider/ServiceProviderClient.java

```diff
@@ -0,0 +1,162 @@
+        int clientReadTimeoutMs = Integer.parseInt(System.getProperty(ZMS_PROP_PROVIDER_READ_TIMEOUT_MS, "15000"));
+
+        return new HttpDriver.Builder("", sslContext)
+                .maxPoolPerRoute(maxPoolRoute)
+                .maxPoolTotal(maxPoolTotal)
+                .clientRetryIntervalMs(clientRetryIntervalMs)
+                .clientMaxRetries(clientMaxRetries)
+                .clientConnectTimeoutMs(clientConnectTimeoutMs)
+                .clientReadTimeoutMs(clientReadTimeoutMs)
+                .hostnameVerifier(new ProviderHostnameVerifier(serviceProviderManager))
+                .build();
+    }
+
+    private DomainDependencyProviderResponse getDependencyStatusFromProvider(ServiceProviderManager.DomainDependencyProvider domainDependencyProvider, String domain, String principal) throws IOException {
+        DomainDependencyProviderResponse domainDependencyProviderResponse = new DomainDependencyProviderResponse();
+        String url = getProviderEndpoint(domainDependencyProvider, domain);
+
+        HttpPost httpPost = new HttpPost(url);
+        httpPost.setHeader("Content-type", "application/json");
+        ProviderDependencyRequest providerDependencyRequest = new ProviderDependencyRequest(
+                "delete",
+                domain,
+                "domain",
+                domain,
+                principal,
+                domainDependencyProvider.getProvider()
+        );
+        String body = jsonMapper.writeValueAsString(providerDependencyRequest);
+        StringEntity stringEntity = new StringEntity(body);
+        httpPost.setEntity(stringEntity);
+        HttpDriverResponse httpResponse = httpDriver.doPostHttpResponse(httpPost);
+        if (httpResponse.getStatusCode() >= 500 && httpResponse.getStatusCode() <= 599) {
+            // provider error - principal should retry operation
+            domainDependencyProviderResponse.setStatus(PROVIDER_RESPONSE_DENY);
+            domainDependencyProviderResponse.setMessage("Http Status: " + httpResponse.getStatusCode() + ", error: " + httpResponse.getMessage());
+        } else if (httpResponse.getStatusCode() >= 400 && httpResponse.getStatusCode() <= 499) {
+            // TODO: Log error and send notification for athenz team on request failure
+        } else if (httpResponse.getStatusCode() >= 200 && httpResponse.getStatusCode() <= 299) {
+            domainDependencyProviderResponse = jsonMapper.readValue(httpResponse.getMessage(), DomainDependencyProviderResponse.class);
+        }
+        return domainDependencyProviderResponse;
+    }
+
+    private String getProviderEndpoint(ServiceProviderManager.DomainDependencyProvider domainDependencyProvider, String domain) {
+        String url = domainDependencyProvider.getProviderEndpoint();
+        if (!url.endsWith("/")) {
+            url += "/";
+        }
+        if (domainDependencyProvider.isInstanceProvider()) {
+            url += "dependency-check/";
+        }
+        url += domain;
```

Comment:
We don't need to include the domainName in the URI. it's a POST request and the payload includes the required domain name.